src/enclave/enclave_signal.c

```diff
@@ -130,24 +130,44 @@ static uint64_t sgxlkl_enclave_signal_handler(
     {
         SGXLKL_TRACE_SIGNAL(
             "Exception SIGILL (illegal instruction) received (code=%d "
-            "address=0x%lx opcode=0x%x)\n",
+            "address=0x%lx rip=0x%lx opcode=0x%x)\n",
             exception_record->code,
             exception_record->address,
+            exception_record->context->rip,
             opcode);
 
         _sgxlkl_illegal_instr_hook(opcode, exception_record->context);
         return OE_EXCEPTION_CONTINUE_EXECUTION;
     }
 
+    /**
+     * Only expose #PF to application signal handler if we permit unsafe host
+     * signals with SGX1.
+     *
+     * With SGX1, there is otherwise no way for the enclave to check if an
+     * in-enclave page fault has actually occured. (For other types of
+     * exceptions, EXITINFO contains trusted execption information inside the
```

Comment:
```suggestion
     * exceptions, EXITINFO contains trusted exception information inside the
```

Reply:
Fixed (and I installed a spell-checker in VSCode).